### ISSUE TITLE
Using DIA-NN results

### ISSUE THREAD

**Issue by Maithy15** [MannLabs/pydiaid#11]
Hi,

Thanks for the nice tool. It would be great if py_diaid could use DIA-NN output as well.

Thanks
Maithy

**Comment by Cajun-data** [comment]
This appears to have been added in the latest version of pyDIAID, however it isn't clear to me which file pyDIAID is expecting from DIA-NN. Furthermore, I am not certain pYDIAID is compatible with DIA-NN 1.9.2 output at the moment. It would be great if a developer could chime in on some of these questions related to using DIA-NN output.

**Comment by Cajun-data** [comment]
There are some clues in the loader_proteomics_library.py function. Specifically:

> dataframe (pd.DataFrame): imported library file from the analysis software
        "DIANN". File format: .csv, required columns: 
        'PrecursorMz',
        'IonMobility',
        'PrecursorCharge',
        'ProteinName',
        'ModifiedPeptide'.
        
Therefore, in R (my preferred language) I can convert a .parquet DIA library to these specifications. 
```
###Convert parquet to csv/tsv

library(arrow)
library(tidyverse)

# Load the Parquet file
df <- read_parquet("DIA_Library.parquet")

#
#"DIANN". File format: .csv, required columns: 
#  'PrecursorMz',
#'IonMobility',
#'PrecursorCharge',
#'ProteinName',
#'ModifiedPeptide'.

df <- df %>%
  rename(IonMobility = IM,
         PrecursorMz = Precursor.Mz,
         PrecursorCharge = Precursor.Charge,
         ProteinName = Protein.Names,
         ModifiedPeptide = Modified.Sequence) %>%
  select(PrecursorMz,IonMobility,PrecursorCharge,
         ProteinName, ModifiedPeptide) %>%
  distinct(ModifiedPeptide, PrecursorCharge, .keep_all = T)

# Save as CSV
write.csv(df, "DIA_Library.csv", row.names = FALSE)
```

**Comment by Maithy15** [comment]
Did the converted format work for you?

**Comment by Cajun-data** [comment]
> Did the converted format work for you?

Yep - the above code works for me. Just make sure to use the right parquet file since there are a few that typically show up in the output. I used the one that corresponds to an experimentally-derived spectral library.
